pkg/cli/login/login.go

```diff
@@ -83,6 +87,8 @@ func NewCmdLogin(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra
 	cmds.Flags().StringVarP(&o.Username, "username", "u", o.Username, "Username for server")
 	cmds.Flags().StringVarP(&o.Password, "password", "p", o.Password, "Password for server")
 
+	cmds.Flags().BoolVarP(&o.WebLogin, "web", "w", o.WebLogin, "Login with web browser")
```

Comment:
As being asked by @atiratree, what is the expected behavior of this flag against 4.13 server?

Reply:
@ardaguclu @atiratree this flag needs the server to have a specific default OAuth client (`openshift-cli-client`) and also the oauth server to have the latest OSIN library; i.e., it needs this code, which will only exist on 4.14 onwards:
- https://github.com/openshift/cluster-authentication-operator/pull/606
- https://github.com/openshift/oauth-server/pull/121
- https://github.com/openshift/oauth-server/pull/128

Any attempt to invoke `oc login --web` against a pre 4.14 server will result on an error on the browser, as the server won't have the necessary oauth client with which this flow proceeds:
```json
{"error":"invalid_request","error_description":"The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed."}
```

Reply:
I would not block on this, but I think it would be better from UX perspective if the user got a preemptive error in the CLI that the web authentication is not possible. It could use the same mechanism as we use in 

https://github.com/openshift/oc/blob/master/pkg/cli/login/loginoptions.go#L292